clients/unshaded/src/main/java/tachyon/client/ClientContext.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import java.net.InetSocketAddress;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.client.block.BSContext;
+import tachyon.client.file.FSContext;
+import tachyon.conf.TachyonConf;
+
+/**
+ * A shared context in each client JVM. It provides common functionality such as the Tachyon
+ * configuration and master address. All members of this class are immutable. This class is
+ * thread safe.
+ */
+public class ClientContext {
+  /**
+   * The static configuration object. There is only one TachyonConf object shared within the same
+   * client.
+   */
+  private static TachyonConf sTachyonConf;
+
+  private static InetSocketAddress sMasterAddress;
+
+  private static UserMasterClientPool sUserClientPool;
+
+  static {
+    sTachyonConf = new TachyonConf();
+
+    String masterHostname = Preconditions.checkNotNull(sTachyonConf.get(Constants.MASTER_HOSTNAME));
+    int masterPort = sTachyonConf.getInt(Constants.MASTER_PORT);
+
+    sMasterAddress = new InetSocketAddress(masterHostname, masterPort);
+
+    sUserClientPool = new UserMasterClientPool(sMasterAddress, sTachyonConf);
+  }
+
+  /**
+   * Returns the one and only static {@link TachyonConf} object which is shared among all classes
```

Comment:
nit: The JavaDoc convention I have been following is to leave out the short description for getters and only use the @return description. So:

@return the Tachyon configuration for the client process